### ISSUE TITLE
Download  Failed ! Please check your internet connection and try again. (5)

### ISSUE THREAD

**Issue by BlackThief1989** [Healix/Gw2Launcher#113]
Guild war 2 works normally but when i try to log in my other accounts from gw2luncher i keep getting this message! : " Download  Failed ! Please check your internet connection and try again. (5) "

please help :(

**Comment by Healix** [comment]
Right click any account > All > Update. This happens when GW2 is patched outside of Gw2Launcher and the Local.dat files still need to be updated.

**Comment by BlackThief1989** [comment]
Thank you <3

**Comment by lukycody** [comment]
I just had the same OP's problem. However, your solution advice did not work for me.

I started the gw2launcher app normally. It then immediately tried to update all login files (20) and failed on each one. I did not update the game via the official game launcher before starting the gw2launcher app btw. 

The only workaround to get the account entries working again was to right click > All > Start normally. The launcher would then consecutively start each account and update, so you'd manually exit each one and let the next one update and so on (takes quite some time with several accounts). However, this also resulted in many login attempts one after another so I also couldn't log with the accounts anymore, getting the login error [undefined].

Any "real" solution for this?
Bonus question: how to prevent those pesky login attempt errors btw.? Also how to stop regularly getting authentication requests, needing to confirm each one?

**Comment by Healix** [comment]
How would "All > Update" fail? Would it have the "failed" status in Gw2Launcher? If so, hover your mouse over it to see why.

Updating first verifies the files used by the account, then launches GW2, so if it failed without launching (it may have flashed for a second if it was already updated), there was something wrong with the files. You could try switching to basic mode under Settings > Guild Wars 2 > Management, which is required if you're not using Windows or have moved %appdata%.

For login errors, there's nothing you can really do unless you can change your IP (it's entirely IP based). There's a hard limit on the number of accounts you can login to (around 15~20), then they temporarily block your IP. If you're not doing that many accounts, there's also a limit on how quickly you can login, which takes longer with every login until it fails - for this, you can enable the limiter under Settings > General > Launching. There's also another lower limit on logins that have SMS enabled; don't use SMS.

For authentication requests, you have to choose to remember your network when GW2 prompts you for the code. It'll do this whenever there's a change in the first 2 or 3 digits of your IP. Gw2Launcher used to automatically handle it in the background, but ArenaNet blocked it a few months ago (if you were using it, disable Settings > Security > Network authorization).

**Comment by lukycody** [comment]
Yes, "All > Update" failed as trying to update and instantly showing the status "fail" on each account entry, without launching. I can't go back and hover over the status to see why, because I already "fixed it" with my workaround that I described above.
Is there some logfile created with such error status where I probably can look that up?
What does basic mode change? Using Windows 10 and I have not moved or changed anything about %appdata%.


About the login/authentication shenanigans: Indeed - having 20 accounts, it's really tedious. Normally when I get to about nr. 16 I run into those errors, needing to double or triple try before finishing the 20th. Oh jeez.
And then, from time to time, I have to go through the pain of confirming each account via email confirmation codes... Wondering if theres something I can configure on my router to prevent this.

**Comment by Healix** [comment]
There's no log for that. If you try to "All > Update" right now, does it fail?

Basic mode changes the location of Local.dat to where GW2 normally has it, whereas virtual mode changes where GW2 looks for Local.dat. If your system didn't support virtual mode, all accounts would share the default Local.dat file regardless of what you configured, but on Windows, it would just fail to launch at all.

**Comment by lukycody** [comment]
Yes, it does fail instantly on all accounts.
Happens both on virtual and basic mode btw.

Any other clues what's wrong or if it's supposed to be like this, as there is no new patch that could be updated to?

**Comment by Healix** [comment]
Right click an account > Selected > Update. I'm assuming it still fails. Hover over that account to see the reason. What does it say?

Also try the [beta version](https://github.com/Healix/Gw2Launcher/blob/master/Gw2Launcher/bin64/Beta/Gw2Launcher.exe).

**Comment by lukycody** [comment]
Alright, tried that again. It says the process needs further rights. So I tried executing the program as administrator and it ran through just fine.

Never had a problem with the program not being ran as administrator before (having only one windows profile which has admin rights). Did you give notice somewhere to always start the program as admin?

**Comment by Healix** [comment]
That's caused by GW2 wanting admin privileges, which happens when GW2 needs to modify its files and is located under a folder that requires elevated privileges to modify, such as Program Files. Launch (normal) should have caused the same error (if there was a patch), but using localized execution will avoid it.

Settings > Guild Wars 2 > Gw2.exe location. Click the "..." button and open GW2. Gw2Launcher will offer to fix it if needed.
